app/controllers/find/courses_controller.rb

```diff
@@ -14,6 +14,8 @@ def show
         subjects: [:financial_incentive],
         site_statuses: [:site]
       ).find_by!(course_code: params[:course_code]&.upcase).decorate
+
+      render_not_found unless @course.is_published?
```

Comment:
Could we tack on `.published` to the `provider.courses` scope rather than handling this separately?
Alternatively should we use the `CourseSearchService` rather than directly accessing Courses 🤔

Reply:
This will be considered as a potential follow up so that we have resolved the issue at hand.